### DISCUSSION TITLE
Question: mux-player without hls.js bundled?

### DISCUSSION THREAD

**Discussion by harshmandan** [muxinc/elements#1033] [Q&A] [answered]
Hello,

We use hls.js library in our app already. But we are experimenting with migrating to using  mux-player. Using the npm package however increases our bundle sizes significantly. I suspect this is because of the bundled hls.js library but we already use hls.js from cdn. Is there a version of mux-player which is decoupled with hls.js and the two libraries then can be loaded separately?

**Comment by luwes** [comment, marked as answer]
Here are all our dist builds:
https://cdn.jsdelivr.net/npm/@mux/mux-player@3.2.0/dist/

Using [import maps](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap) you should be able to direct the package imports to the urls of your choice.

I put together an example here: https://codesandbox.io/p/sandbox/8xgx86?file=%2Findex.html%3A17%2C77

Note that I wouldn't use this in production because of the many package requests it has to make.

---

A better solution would be to NPM install Mux player in a project, then import the `/dist/index.mjs` path in your entry file and make the `hls.js` import an external package (https://esbuild.github.io/api/#external). This will make that import not be resolved and not included in the bundle. Bundle everything, host the bundle somewhere. Then use an importmap for just the `hls.js` import...

Hope this helps!

**Reply by harshmandan** [reply]
@luwes I have since moved on to using the simpler `<mux-video/>` element. I tried doing this with mux-video, but it seems like hls.js can't be externalized when using `mux-video`. Looking at the package installed at:`/node_modules/@mux/mux-video/dist/mux-video.js` - I can see that hls.js is part of the same js file. I tried setting up the external config in vite and tried using import maps but the post-built bundle sizes remains same, not sure what am I doing wrong?

**Reply by luwes** [reply]
try using this unbundled dist file https://cdn.jsdelivr.net/npm/@mux/mux-video@0.26.1/dist/index.mjs or node_modules/@mux/mux-video/dist/index.mjs

**Reply by harshmandan** [reply]
Hey @luwes, I tried making externalization work but can't seem to get it work. 
I've gotten it to work with import maps but as you mentioned it sends a lot of network requests. 
The goal is to externalize hls.js as it is used in other parts of the app. 

Here's what I've tried: https://github.com/harshmandan/externalize-mux-video

---

Edit: Got it to work, the config is pretty simple. I've pushed the updated config in the same repo!
